share/sql/sql.go

```diff
@@ -613,133 +500,49 @@ func (m *mgr) GetReceivedShare(ctx context.Context, ref *collaboration.ShareRefe
 }
 
 func (m *mgr) UpdateReceivedShare(ctx context.Context, share *collaboration.ReceivedShare, fieldMask *field_mask.FieldMask) (*collaboration.ReceivedShare, error) {
+
 	user := appctx.ContextMustGetUser(ctx)
 
 	rs, err := m.GetReceivedShare(ctx, &collaboration.ShareReference{Spec: &collaboration.ShareReference_Id{Id: share.Share.Id}})
 	if err != nil {
 		return nil, err
 	}
 
-	for i := range fieldMask.Paths {
-		switch fieldMask.Paths[i] {
+	shareState, err := m.getShareState(ctx, share.Share.Id, user)
+	if err != nil {
+		return nil, err
+	}
+
+	// FieldMask determines which parts of the share we actually update
+	// Right now, only updating the state is supported
```

Comment:
Update comment.

Reply:
I would also explain why we have this "state" and "hidden" since it's not clear to someone unaware of the legacy state.